src/api/product/useFetch/useFetch.ts

```diff
@@ -0,0 +1,27 @@
+import { useEffect, useState } from 'react';
+import { FetchReturnedValue } from './useFetch.types';
+
+export const useFetch = <Type>(url: string): FetchReturnedValue<Type> => {
+  const [isError, setIsError] = useState(false);
+  const [data, setData] = useState<Type | undefined>(undefined);
+  const isLoading = !data && !isError;
```

Comment:
Ten loading będzie działał tylko za pierwszym razem, jeśli będzie trzeba zrobić refetch, bo np zmienił się url, to data zawsze będzie występować. Jak myślisz, jak to można rozwiązać inaczej (nie używając dodatkowego `useState`)?

Reply:
Trzeba ustawić data i isError na wartości domyślne w `cleanup function useEffect-a`. Urla mamy w `dependencies` useEffect-a`, więc zanim `useFetch` wywoła `fetch`, `isError, data, isLoading` będą miały wartości domyślne.